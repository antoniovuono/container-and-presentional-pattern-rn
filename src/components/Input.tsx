import { StyleSheet, TextInput, TextInputProps } from 'react-native';

export function Input({ ...rest }: TextInputProps) {
  return (
    <TextInput
      {...rest}
      style={styles.input}
      placeholder="username"
      placeholderTextColor="#C4C4CC"
      autoCapitalize="none"
    />
  );
}

const styles = StyleSheet.create({
  input: {
    backgroundColor: '#202024',
    padding: 15,
    borderRadius: 8,
    color: '#FFFFFF',
    width: '100%',
  },
});
