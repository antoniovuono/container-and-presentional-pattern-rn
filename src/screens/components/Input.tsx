import { StyleSheet, TextInput } from 'react-native';

export function Input() {
  return (
    <TextInput
      style={styles.input}
      placeholder="username"
      placeholderTextColor="#C4C4CC"
      autoCapitalize="none"
    />
  );
}

const styles = StyleSheet.create({
  input: {
    backgroundColor: '#202024',
    padding: 15,
    borderRadius: 8,
    color: '#FFFFFF',
    width: '100%',
  },
});
