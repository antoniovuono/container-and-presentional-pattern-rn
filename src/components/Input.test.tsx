import { render } from '@testing-library/react-native';
import { Input } from './Input';

describe('Input component', () => {
  it('should be able to render component correctly', () => {
    const tree = render(<Input />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
