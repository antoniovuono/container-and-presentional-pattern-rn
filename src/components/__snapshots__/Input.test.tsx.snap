// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should be able to render component correctly 1`] = `
<TextInput
  autoCapitalize="none"
  placeholder="username"
  placeholderTextColor="#C4C4CC"
  style={
    {
      "backgroundColor": "#202024",
      "borderRadius": 8,
      "color": "#FFFFFF",
      "padding": 15,
      "width": "100%",
    }
  }
/>
`;

exports[`Input component should be able to render componente correctly 1`] = `
<TextInput
  autoCapitalize="none"
  placeholder="username"
  placeholderTextColor="#C4C4CC"
  style={
    {
      "backgroundColor": "#202024",
      "borderRadius": 8,
      "color": "#FFFFFF",
      "padding": 15,
      "width": "100%",
    }
  }
/>
`;
